src/modules/flight_mode_manager/tasks/ManualPositionHeadless/FlightTaskManualPositionHeadless.hpp

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file FlightTaskManualPositionHeadless.hpp
+ *
+ * Flight task for manual controlled headless position.
+ *
+ * @author Inhwan Wee <inhwan.wee@gmail.com>
+ */
+
+#pragma once
+
+#include <lib/collision_prevention/CollisionPrevention.hpp>
+#include "FlightTaskManualPosition.hpp"
+
+class FlightTaskManualPositionHeadless : public FlightTaskManualPosition
```

Comment:
Note that you are creating a complete copy of the `ManualPosition` task which generates unsmooth velocity setpoints directly from stick and will result in quite jerky position mode for a beginner. That task is only kept for velocity controller tuning.

Instead, you could inherit from `FlightTaskManualAcceleration` and just pass your reference yaw saved at the beginning instead of `_yaw_setpoint` here: https://github.com/PX4/PX4-Autopilot/blob/2d15c96b9b3aad71b9fcb1333aa30cb280177c62/src/modules/flight_mode_manager/tasks/ManualAcceleration/FlightTaskManualAcceleration.cpp#L72

I'm assuming that does what you want.